### PR TITLE
Move game logic to game class

### PR DESCRIPTION
**Pull request by effectpet**

### CODE REVIEW

File: lib/App.js [hunk 1]
```diff
@@ -45,40 +31,19 @@ module.exports = class App {
...
-
 	onDiscordReady() {
 		console.info("Discord bot is ready");
 	}
 
 	onDiscordMessage(message) {
-		if (this.isMessageSendByThisBot(message)) {
-			this.lastMessageOfBot = message;
+		if (this.isMessageSendByThisBot(message) === true && this.game !== null) {
+			this.game.lastMessageOfBot = message;
```
Comment (Absebo) [inline lib/App.js]: this.game.onLastMessageOfBot(message)

File: lib/App.js [hunk 2]
```diff
@@ -128,97 +93,8 @@ module.exports = class App {
...
-		}
-	}
-
-	gameAddLength() {
-		this.currentLength += 1;
-	}
-
 	onAdd(message) {
-		if (this.gameState !== GameStateEnum.NOT_RUNNING) {
+		if (this.game !== null) {
```
Comment (Absebo) [inline lib/App.js]: this could be a function.... like `this.isGameRunning()`

File: lib/App.js [hunk 3]
```diff
@@ -267,33 +142,41 @@ module.exports = class App {
...
-
-		this.gameState = GameStateEnum.STARTING;
-		this.currentChannel = message.channel;
+		this.game = new Game(
+			this.translation,
+			this.gameMode,
+			message.channel,
+			this.players,
+		);
+		this.game.on('stop', this.resetGame.bind(this));
```
Comment (Absebo) [inline lib/App.js:151]: this.game.onStop(this.resetGame.bind(this));

File: lib/App.js [hunk 4]
```diff
@@ -267,33 +142,41 @@ module.exports = class App {
...
 
-		message.channel.send(this.translation.f('game.gameStared', {username: message.author.username}));
-		this.callNext();
+		message.channel.send(this.translation.f('game.gameStared', { username: message.author.username }));
 	}
 
 	onStop(message) {
-		// Check if no game is running
-		if (this.gameState === GameStateEnum.NOT_RUNNING) {
+		if (this.game === null) {
```
Comment (Absebo) [inline lib/App.js]: !this.isGameRunning()

File: lib/Game.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+		this.sleepTimeout = setTimeout(this.startNewRound.bind(this), TIMEOUT_SLEEP);
+	}
+
+	startNewRound() {
+		// Create new message
+		this.currentString = this.randomGenerator.getRandomString(this.currentLength);
+
+		// Get random player & send new message
+		const randomPlayerIndex = Math.floor(Math.random() * this.currentPlayers.length);
+		const randomPlayer = this.currentPlayers.splice(randomPlayerIndex, 1);
```
Comment (Absebo) [inline lib/Game.js:87]: please move this logic to a function 
`this.currentPlayer = this.getRandomPlayer();`

File: lib/Game.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+			if (this.gameMode === GameModeEnum.SPEED) {
+				this.gameAddSpeed();
+			} else if (this.gameMode === GameModeEnum.LENGTH) {
+				this.gameAddLength();
+			}
+		}
+		this.firstStart = true;
+
+		// Send new round info
+		this.channel.send(this.translation.f('game.newRound', { currentSpeed: this.currentSpeed })); // + this.getCurrentInfo());
```
Comment (Absebo) [inline lib/Game.js]: `// + this.getCurrentInfo());` dead code

**Review by Absebo — CHANGES_REQUESTED** [review]

File: lib/Game.js (hunk 5 above)
Comment (effectpet) [reply lib/Game.js:87]: The whole function is about getting a new player and send him an message. I could rename it, but creating a new function is stupid.

File: lib/App.js (hunk 3 above)
Comment (effectpet) [reply lib/App.js:151]: There is no real benefit. Keep it simple

**Review by Absebo — APPROVED** [review]

**Pull request merged**